### PR TITLE
LPD-24644 Create ref to publish tag to github

### PR DESCRIPTION
**Pull request by lucasmiranda0**
Related issue: [LPD-24644](https://liferay.atlassian.net/browse/LPD-24644)

### CODE REVIEW

File: release/release_gold.sh [hunk 1]
```diff
@@ -110,23 +110,58 @@ function tag_release {
...
-			--request POST \
-			--retry 3 \
-			--silent)
+	local username=liferay
+
+	if [ -n "${LIFERAY_RELEASE_USER_REPOSITORY}" ]
+	then
+		username=${LIFERAY_RELEASE_USER_REPOSITORY}
+	fi
+
```
Comment (natocesarrego) [inline release/release_gold.sh:119]: @lucasmiranda0, why did you need this? Can you please clarify?

File: release/release_gold.sh [hunk 2]
```diff
@@ -110,23 +110,58 @@ function tag_release {
...
+			}' \
+		--fail \
+		--header "Accept: application/vnd.github+json" \
+		--header "Authorization: Bearer ${LIFERAY_RELEASE_GITHUB_PAT}" \
+		--header "X-GitHub-Api-Version: 2022-11-28" \
+		--max-time 10 \
+		--request POST \
+		--retry 3 \
+		--silent \
+		| awk '/^HTTP/{print $2}')
```
Comment (natocesarrego) [inline release/release_gold.sh:137]: @lucasmiranda0, can you please give me more details about this code? Why do you need this?

File: release/release_gold.sh [hunk 3]
```diff
@@ -110,23 +110,58 @@ function tag_release {
...
+	then
+		username=${LIFERAY_RELEASE_USER_REPOSITORY}
+	fi
+
+	local response_tag_creation=$(curl -i \
+		"https://api.github.com/repos/${username}/${repository}/git/tags" \
+		--data-raw '
+			{
+				"message": "",
+				"object": "'$(lc_get_property release-data/release.properties git.hash.liferay-portal-ee)'",
```
Comment (natocesarrego) [inline release/release_gold.sh:125]: @lucasmiranda0, will the path `release-data/release.properties` still work? Don't we need to use the absolute path instead, like in https://github.com/liferay/liferay-docker/blob/73809f47af89ccb12d0d38ebe4cd0bf333cb0b4e/release/_git.sh#L174?

File: release/release_gold.sh [hunk 4]
```diff
@@ -110,23 +110,58 @@ function tag_release {
...
+
+	if [ "${response_tag_creation}" -ge 300 ]
+	then
+		lc_log ERROR "Unable to tag release."
+
+		return "${LIFERAY_COMMON_EXIT_CODE_SKIPPED}"
+	fi	
+
+	local response_ref_creation=$(curl -i \
+		"https://api.github.com/repos/${username}/${repository}/git/refs" \
```
Comment (natocesarrego) [inline release/release_gold.sh:147]: @lucasmiranda0, why do you need this extra API call? Can you please clarify?

Thanks

**Review by natocesarrego — COMMENTED** [review]
@lucasmiranda0, please check my comments, rebase to fix conflicts and re-send.

Thanks

File: release/release_gold.sh (hunk 1 above)
Comment (lucasmiranda0) [reply release/release_gold.sh:119]: I was already thinking on the test perspective. If we don't have a variable like that, every time that we test we have to change the user in the code, otherwise we won't create a tag, unless you have permission to create a tag to the Liferay repository.

File: release/release_gold.sh (hunk 3 above)
Comment (lucasmiranda0) [reply release/release_gold.sh:125]: Forgot to change that.

File: release/release_gold.sh (hunk 2 above)
Comment (lucasmiranda0) [reply release/release_gold.sh:137]: The previous verification was not working well on my tests, so I started to verify based on the HTTP response. But I can recheck the previous behavior.

File: release/release_gold.sh (hunk 4 above)
Comment (lucasmiranda0) [reply release/release_gold.sh:147]: It was needed to publish the tag to the GitHub. That API call creates the reference needed. I made based on this [GitHub article](https://docs.github.com/en/rest/git/tags?apiVersion=2022-11-28#create-a-tag-object).
